lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -99,7 +101,29 @@ def scrub(transcript)
           gsub(/("number\\":\\")\d+/, '\1[FILTERED]').
           gsub(/("cvv\\":\\")\d+/, '\1[FILTERED]').
           gsub(/("cryptogram\\":\\")\w+/, '\1[FILTERED]').
-          gsub(/(source\\":\{.*\\"token\\":\\")\d+/, '\1[FILTERED]')
+          gsub(/(source\\":\{.*\\"token\\":\\")\d+/, '\1[FILTERED]').
+          gsub(/("token\\":\\")\w+/, '\1[FILTERED]')
+      end
+
+      def store(payment_method, options = {})
+        post = {}
+        MultiResponse.run do |r|
+          response = r.process { tokenize(payment_method, options) }
+          if response.success?
```

Comment:
💬 Comment:

Just to prevent if nesting, what do you think about something like this:

```Ruby
MultiResponse.run do |r|
  response = r.process { tokenize(payment_method, options) }
  break unless response.success?

  token = response.params['token']
  add_payment_method(post, token, options)
  add_customer_data(post, options)
  post = post.merge(post.delete(:source))
  r.process { commit(:store, post) }
end
```